### DISCUSSION TITLE
Events: persistent or transient

### DISCUSSION THREAD

**Discussion by tfahringer** [alexzhornyak/SCXML-tutorial#25] [Q&A]
If an event is raised either internally or by external sources, are events lost if not catched by an active state or are they stored somewhere until catched by a state or transition?
What is the semantic of that under SCXML

**Comment by alexzhornyak** [comment]
From the docs:

The SCXML Processor must bind the [**_event** variable](https://www.w3.org/TR/scxml/#SystemVariables) when an event is pulled off the internal or external event queue to be processed, and must keep the variable bound to that event until another event is processed. 

So, only last event is binded to global datamodel variable **_event**, but note that user events can be replaced with scxml internal 'done.state', 'done.invoke' events, etc.

![image](https://user-images.githubusercontent.com/18611095/233656949-bf9c66fc-0f07-417f-817b-0098d9e555c5.png)

![image](https://user-images.githubusercontent.com/18611095/233657434-c678915b-6936-4005-ae41-a7c68d39dfb8.png)
